### ISSUE TITLE
Player inventory editor

### ISSUE THREAD

**Issue by technoguy20000** [Bertware/bukkitgui2#50]
An option like this in the players tab would be useful.

**Comment by Bertware** [comment]
While this would be useful, this is close to impossible. I can't alter the player data files while the server is running, and using a plugin would make it version and server dependant.

**Comment by technoguy20000** [comment]
There are plugins that can do this, but don't have an interface like the bukkitGUI has the potential to. Perhaps an added plugin could allow for this to happen.

**Comment by technoguy20000** [comment]
I don't see a good way to do this aside from making several plugin versions, or somehow making a plugin that is universally compatible with all versions. I will reopen this if I find a better way to do it.

**Comment by spideynn** [comment]
@Bertware You could use the give command instead of directly modifying the player data files, although that might be dependent on the server software.

**Comment by Bertware** [comment]
That would work, but it's limited to giving items only, not viewing or editing the inventories (which is what @technoguy20000 meant)

**Comment by technoguy20000** [comment]
I've heard from a plugin developer that the plugin is not version specific if it doesn't have version specific imports, in other words, using only the BukkitAPI

**Comment by technoguy20000** [comment]
Just by the way, the give command paired with the clear command (/clear [player] [item] [data] [maxCount] [dataTag]) and the .dat file for viewing the inventory, this approach still could work.

**Comment by spideynn** [comment]
I was thinking about making a plugin for it, since I figured out a way that shouldn't depend on a certain version. How should the information be transferred between the plugin and BukkitGUI?

**Comment by Bertware** [comment]
The .dat file isn't up to date while the server is running I believe, or it's locked. It might be possible though, although it's not my number one priority.
A possible solution would be this:
* User clicks "view inventory"
* GUI sends command "/bukkitguihelp inventory <username> (command limited to console operator only)
* GUI receives JSON encoded array, hides the output from the user
* GUI parses JSON and shows window

**Comment by simon816** [comment]
I vote a [Sponge](http://spongepowered.org) plugin, since Bukkit will be obsolete in a few months.

It should be an optional extra though.

**Comment by spideynn** [comment]
@Bertware I tested this on my server, you can modify the .dat while the server is running but the player needs to disconnect before the changes take effect, or they get overwritten when something happens.
@simon816 I'll do both, but I need to learn the Sponge API first. I know that Glowstone uses the Bukkit API

**Comment by Bertware** [comment]
Before y'all start working on plugins or stuff: both glowstone and sponge support is planned or work in progress, but as far as I know they both don't work 100% as they should now. 
Just a reminder, so you don't get disappointed when you want to test

**Comment by spideynn** [comment]
I've almost finished the plugin, and here's the output when you do '/bukkitguihelp inventory <playername>' http://pastebin.com/L8DhNGad
I'm just trying to figure out why its adding (0) on the end of the item name, and after that the bukkit / spigot plugin is done once that is fixed.

**Comment by spideynn** [comment]
Spigot, Bukkit, and Glowstone support for only reading the inventory is available here: https://github.com/spideynn/InventoryAPI-JSON/releases/

**Comment by Bertware** [comment]
I'm currently fixing up the server support before I'm continuing on this myself (if you know .NET, feel free to create an inventory edit window that takes the console output and player name, shows the inventory, and sends the necessary commands to alter the inventory)

**Comment by spideynn** [comment]
Yeah, I'll see what I can do with it. I just need to look through the codebase to learn how you do things in this.

**Comment by Bertware** [comment]
You could add it to the MinecraftInterop namespace, a class and a form to edit inventories. I try to put all the forms in the UI folder/namespace, thought addon specific forms are in the corresponding addon folder/namespace. Since this would be globally available in both console and player list, I'd keep it as a global thing
